src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -5922,12 +5943,24 @@ private void ParseTypeArgumentList(out SyntaxToken open, SeparatedSyntaxListBuil
             // remaining types & commas
             while (true)
             {
-                if (this.CurrentToken.Kind == SyntaxKind.GreaterThanToken)
-                {
+                if (this.CurrentToken.Kind is SyntaxKind.GreaterThanToken)
                     break;
-                }
-                else if (this.CurrentToken.Kind == SyntaxKind.CommaToken || this.IsPossibleType())
+
+                // We prefer parsing as if there was a missing > over parsing as a tuple type for error recovery
+                if (tokenKindBreaksTypeArgumentList(this.CurrentToken.Kind))
+                    break;
+
+                if (this.CurrentToken.Kind == SyntaxKind.CommaToken || this.IsPossibleType())
                 {
+                    // we are currently past parsing a type and we encounter an unexpected identifier token
+                    // followed by tokens that are not part of a type argument list
+                    // example: List<(string a, string b) Method() { }
+                    //                 current token:     ^^^^^^
+                    if (this.CurrentToken.Kind is SyntaxKind.IdentifierToken && tokenKindBreaksTypeArgumentList(this.PeekToken(1).Kind))
+                    {
+                        break;
+                    }
```

Comment:
this shoudl be broken out as well.  Specifically, i would not run in this in the comma case, only in the "this.IsPossibleType" case.